src/network-events.c

```diff
@@ -52,6 +52,47 @@ static __always_inline int save_sock_ptr(struct pt_regs *ctx, void *map)
     return 0;
 }
 
+typedef struct
+{
+    u16 protocol_type;           // Something like IPPROTO_TCP or IPPROTO_UDP
+    u32 pid;
+    u16 remote_port;
+    ip_addr_t protos;
```

Comment:
nit: we should probably order this in ascending size so the struct has less gaps and takes less memory.